contributors/design-proposals/support-union-in-api-server.md

```diff
@@ -0,0 +1,200 @@
+Add Tag Indicating To Replace *Union* Fields in Patch
+=============
+
+Add a tag `patchStrategy:"replaceKey"`. For a given type that has this tag, all keys/fields missing 
+from the request will be cleared when patching the object. For a field presents in the request,
+it will be merged with the live config.
+
+This approach could be used to fix the issues with `kubectl apply` (e.g. [#34292](https://github.com/kubernetes/kubernetes/issues/34292))
+
+The proposal of Full Union is in [kubernetes/community#388](https://github.com/kubernetes/community/pull/388).
+
+| Capability | Supported By This Proposal | Supported By Full Union |
+|---|---|---|---|
+| Auto clear missing fields on patch | X | X |
+| Merge union fields on patch | X | X |
+| Validate only 1 field set on type |  | X |
+| Validate discriminator field matches one-of field |  | X |
+| Support non-union patchKey | X | TBD |
+| Support arbitrary combinations of set fields | X |  |
+
+## Proposed Changes
```

Comment:
Clarify the behavior for each of the new tags and old tags.  e.g. mergeKeys must always be present to perform the merge, and will be preserved.